Shared/Transforms/Variable.cs

```diff
@@ -0,0 +1,334 @@
+        public Variable(string name, object value)
+        {
+            if(!CheckName(name))
+                throw new ArgumentException("The name can include only alphanumeric characters and must begin with a letter.");
+
+            m_name = name;
+            m_value = value ?? throw new ArgumentException("The value cannot be null."); 
+        }
+
+        /// <summary>
+        /// User defined variable constructor by name and Expression.
+        /// </summary>
+        /// <param name="name">Variable name.</param>
+        /// <param name="value">The Expression object by which the variable will be initialized.</param>
+        public Variable(string name, Expression value)
+        {
+            if (!CheckName(name))
+                throw new ArgumentException("The name can include only alphanumeric characters and must begin with a letter.");
+                            
+            m_name = name;
+            m_value = value ?? throw new ArgumentException("The value cannot be null.");
+        }
+
+        /// <summary>
+        /// User defined variable constructor by name and string list.
+        /// </summary>
+        /// <param name="name">Variable name.</param>
+        /// <param name="multipleStringValues">The string list by which the variable will be initialized.</param>
+        public Variable(string name, List<string> multipleStringValues)
+        {
+            if (!CheckName(name))
+                throw new ArgumentException("The name can include only alphanumeric characters and must begin with a letter.");
+
+            m_name = name;
+
+            if (multipleStringValues != null)
+                m_value = string.Join(":", multipleStringValues);
+            else
+                m_value = string.Empty;
+        }
+
+        private bool CheckName(string name)
+        {
+            if (string.IsNullOrWhiteSpace(name))
+                return false;
+
+            if (char.IsDigit(name[0]))
+                return false;
+
+            Regex alphanumericPattern = new Regex("^[a-zA-Z0-9]*$");
+
+            return alphanumericPattern.IsMatch(name);
+        }
+
+        public string Key
+        {
+            get
+            {
+                return string.Format("${0}", m_name);
+            }
+        }
+
+        public string TextLayerKey
+        {
+            get
+            {
+                return string.Format("$({0})", m_name);
+            }
+        }
+
+        public string Value
+        {
+            get
+            {
+                return GetVariableValue();
+            }
+        }
+
+        public override string ToString()
+        {
+            return string.Format("${0}_{1}", m_name, GetVariableValue());
+        }
+
+        private string GetVariableValue()
+        {
+            string result = m_value.ToString();
+
+            var @types = new Dictionary<Type, Action> {
```

Comment:
why do we need this dictionary?
the "supported" types are listed here and ToSTring should work for all. the only special case is the string type (which can be handled before calling the generic ToString())
in addition if the type of the object is not supported @types[m_value.GetType()](); will throw an exception.
if we support small subset of types we should validate them in the construction phase.

Reply:
We prefer to keep this, if possible. The standard method ToString() operates in alignment with regional special characteristics. For instance, when converting float type we can get two different values, like "2.0" and "2,0", depending on client's location (System.Globalization.CultureInfo). Cloudinary's server accepts only one of these, so what we can maybe do is:
1. Leave dictionary;
2. Add a code when converting into a string for each one of the types (wherever necessary) listed in the dictionary. This code could've excluded regional special characteristics and would've always returned the same result according to Cloudinary's server.

Reply:
So can't we have only the first ToString with the same exclusion without keeping the dictionary?

Reply:
Yup, we'll do that. Just waiting for your reply on my other question here to do it all together.